hr_attendance_reason/static/src/js/kiosk_confirm.esm.js

```diff
@@ -0,0 +1,77 @@
+/** @odoo-module **/
+import KioskConfirm from "hr_attendance.kiosk_confirm";
+import {_t} from "web.core";
+
+const superSignInOut =
+    KioskConfirm.prototype.events["click .o_hr_attendance_sign_in_out_icon"];
+const superPinPadButton =
+    KioskConfirm.prototype.events["click .o_hr_attendance_pin_pad_button_ok"];
+
+KioskConfirm.include({
+    events: Object.assign({}, KioskConfirm.prototype.events, {
+        "click .o_hr_attendance_sign_in_out_icon": function () {
+            this.update_attendance(superSignInOut);
+        },
+        "click .o_hr_attendance_pin_pad_button_ok": function () {
+            this.update_attendance(superPinPadButton);
+        },
+    }),
+    willStart: async function () {
+        await this._super();
+        await this._rpc({
+            model: "hr.employee",
+            method: "search_read",
+            args: [
+                [["user_id", "=", this.getSession().uid]],
+                [
+                    "show_reasons_on_attendance_screen",
+                    "required_reason_on_attendance_screen",
+                ],
+            ],
+        }).then((res) => {
+            this.employee = res.length && res[0];
+        });
+        await this._rpc({
+            model: "hr.attendance.reason",
+            method: "search_read",
+            fields: ["name", "action_type"],
+            domain: [["show_on_attendance_screen", "=", true]],
+        }).then((reasons) => {
+            this.reasons = reasons;
+        });
+    },
+    _rpc: function () {
```

Comment:
Let's comment this override as the way to inject into the context of an enclosed rpc call